guava/src/com/google/common/primitives/Ints.java

```diff
@@ -505,6 +506,35 @@ public static void reverse(int[] array, int fromIndex, int toIndex) {
       array[j] = tmp;
     }
   }
+  
+  /**
+   * Shuffles elements of {@code array}. This is equivalent to {@code
+   * Collections.shuffle(Ints.asList(array))}, but is likely to be more efficient.
+   *
+   */
+  public static void shuffle(int[] array) {
+    checkNotNull(array);
+    shuffle(array, 0, array.length);
+  }
+
+  /**
+   * Shuffles elements of {@code array} between {@code fromIndex} inclusive and {@code toIndex}
+   * exclusive. This is equivalent to {@code
+   * Collections.shuffle(Ints.asList(array).subList(fromIndex, toIndex))}, but is likely to be more efficient.
+   *
+   * @throws IndexOutOfBoundsException if {@code fromIndex < 0}, {@code toIndex > array.length}, or
+   *    {@code toIndex > fromIndex}
+   */
+  public static void shuffle(int[] array, int fromIndex, int toIndex) {
+    checkNotNull(array);
+    checkPositionIndexes(fromIndex, toIndex, array.length);
+    for (int i = fromIndex, j = toIndex - 1; i < j; i++, j--) {
+      int randomPosition = ThreadLocalRandom.current().nextInt(array.length);
```

Comment:
- The `randomPosition` should probably be between 0 and `toIndex` instead of the array length?
- Are you using `j` only to check the condition `i < j`? It's not used in the code. In general I wonder if this really randomly shuffles all parts of the array the same or if some parts are more prone to staying in the same area...